app/models/practice_editor_session.rb

```diff
@@ -0,0 +1,178 @@
+    last_name = user.last_name.nil? ? "?" : user.last_name
+    ret_val = first_name + " " + last_name
+    ret_val += " @" + rec.session_start_time.to_s if include_time
+    ret_val
+  end
+  def self.locked_by_user(rec_id)
+    rec = PracticeEditorSession.find_by_id(rec_id)
+    user_id = rec.user_id
+    user = User.find_by_id(user_id)
+    return user.blank? ? 0 : user.id
+  end
+  def self.is_admin(user_id)
+    sql = "select r.name from roles r join users_roles ur on r.id = ur.role_id join users u on ur.user_id = u.id where u.id = $1"
+    param1 = "#{user_id}"
+    records_array = ActiveRecord::Base.connection.exec_query(sql, "SQL", [[nil, param1]])
+    if records_array.count == 1 then
+      if records_array[0]["name"] == "admin" then
+        return true
+      end
+    end
+    return false
+  end
+  def self.practice_last_updated(practice_id)
+    session = PracticeEditorSession.where(practice_id: practice_id).where.not(session_end_time: nil).order("session_end_time DESC").first()
+    return "" if session.blank?
+    lock_user = locked_by(session.id, false)
+    s_text = "Practice last updated on " + session.session_end_time.strftime("%B %d, %Y") + " at " + session.session_end_time.strftime("%I:%M %p").gsub(/^0/,'')
+    s_text += " by " +  lock_user
+    if is_admin(session.user_id)
+      s_text += " (Site Admin)"
+    end
+    s_text
+  end
+
+  def self.monitor_editing_session(practice_id, user_id)
+    #TODO update session record with process ID...
+    session_rec = PracticeEditorSession.where(practice_id: practice_id, user_id: user_id, session_end_time: nil, process_id: nil).order("session_start_time DESC").first()
+    Thread.new do
+      if !session_rec.blank?
+        session_rec.process_id = Thread.current.object_id.to_i
+        session_rec.save
+      else
+        Thread.exit
+        return
+      end
+        monitor_session(practice_id, user_id, session_rec.id)
+    end
+  end
+  def self.monitor_session(practice_id, user_id, session_rec_id)
+    loop do
+      session_rec = PracticeEditorSession.where(practice_id: practice_id, user_id: user_id, session_end_time: nil).order("session_start_time DESC").first()
+      if session_rec.blank?
+        Thread.exit
+      end
+      sleep 50
+      diff = minutes_in_session(session_rec.session_start_time)
+      is_published = Practice.find_by_id(practice_id).published
+      # puts 'session_start_time: ' + session_rec.session_start_time.to_s
+      # puts 'diff from now: ' + diff.to_s
+      # puts 'rec_id: ' + session_rec_id.to_s
+      # puts 'process id: ' + session_rec.process_id.to_s
+      # puts 'thread_id: ' + Thread.current.object_id.to_s
+      # puts practice_id.to_s + ", " + user_id.to_s
+      #TODO: set back to diff > 14 .set to 0 only for testing.. bj_2_10_2021
+      if (is_published && diff > 14) || (!is_published && diff > 29)
+        rec = PracticeEditorSession.find_by_id(session_rec_id)
+        if !rec.blank?
```

Comment:
I think we can use `rec.present?` here instead of `!.rec.blank?`.

Reply:
done

Reply:
I updated the var name below from ctr to session_counter